modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/Schemas.java

```diff
@@ -341,6 +354,55 @@ public static final XSDSchema parse(String location, ResourceSet resourceSet)
         }
     }
 
+    /**
+     * Fetches the contents of the URI into a local {@link InputStream}.
+     * 
+     * @param uri
+     * @param options
+     * @param response
+     * @return A local resource
+     * @throws IOException
+     */
+    private static InputStream readUriResource(URI uri, ResourceSet resourceSet, Map<?, ?> response)
```

Comment:
Could you rework this one so that it uses try-with-resources and returns a byte[], which makes evident it's stored in memory and there are no leftovers to close?